compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BFiniteType.java

```diff
@@ -96,8 +96,12 @@ public boolean isNullable() {
 
     public void addValue(BLangExpression value) {
         this.valueSpace.add(value);
-        if (!nullable && value.getBType().isNullable()) {
-            nullable = true;
+        if (!this.nullable && value.getBType() != null &&  value.getBType().isNullable()) {
+            this.nullable = true;
         }
     }
+
+    public void setValueSpace(Set<BLangExpression> valueSpace) {
+        this.valueSpace = valueSpace;
```

Comment:
This is okay only if this is done at ctor right. 
Lets say we do this for a `nullable` `BFiniteType` type. But due to the new set now the `BFiniteType` is not nullable but we forgot to change nullable status.

Reply:
Can you give an example where nullable becomes 'true' in `BFiniteType`